src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -141,7 +150,20 @@ public int getCentralTagId() {
         // make a temp holder var for least Y translation, set to first tags translation
         double minY = 1.0e6; // big number
         int targetID = -1;
-        for (PhotonTrackedTarget tag : targetResult.getTargets()) { // for every target in camera
+        for (PhotonTrackedTarget tag : targetResult.getTargets()) { // for every target in camera            
+            // find id for current tag we are focusing on
+            int tempTagID = tag.getFiducialId();
+
+            // if aiming for substation
+            if (substationTarget) {
+                if (tempTagID != 5 || tempTagID != 4) { //exit if tags are grids
+                    break;
+                }
+            } else { //if aiming for aiming for grid/not substation
+                if (tempTagID == 5 || tempTagID == 4) { // exit if tag is substation
```

Comment:
It would be cleaner if you first create a variable like "isSubstationTarget", and then do the if/else tests.